### PR TITLE
Refactoring: use observable.pipe(map) instead of observable.map, etc.

### PR DESCRIPTION
**Pull request by romanroe**

### CODE REVIEW

File: frontend/app/components/wp-relations/wp-relations.directive.ts [hunk 1]
```diff
@@ -70,18 +73,12 @@ export class WorkPackageRelationsController {
       });
   }
 
-  protected getRelatedWorkPackages(workPackageIds:string[]) {
-    let observablesToGetZipped = workPackageIds.map(wpId => {
+  private getRelatedWorkPackages(workPackageIds:string[]):Observable<WorkPackageResourceInterface[]> {
```
Comment (romanroe) [inline frontend/app/components/wp-relations/wp-relations.directive.ts:76]: @oliverguenther The old code was a bit weird. Maybe you can double-check my changes?

Comment (oliverguenther) [reply frontend/app/components/wp-relations/wp-relations.directive.ts:76]: @romanroe You need to call zip with the observables as arguments, not as an array. I've fixed this commit that should resolve the remaining integration tests.

**Review by oliverguenther — CHANGES_REQUESTED** [review]
Karma tests are still failing with weird messages that `.pipe()` is not a function. Can you take a look at it? I couldn't find out why that happens.

That being said, I fixed an error in the `zip` code that caused backend integration tests to fail. If npm is fixed, good to merge from my side.

**Comment by romanroe** [comment]
@oliverguenther IMHO ready to merge :-)

**Review by oliverguenther — APPROVED** [review]

**Pull request merged**